### PR TITLE
FEATURE: smart date support for AI helper

### PR DESCRIPTION
**Pull request by SamSaffron**
This feature allows conversion of human typed in dates and times
to smart "Discourse" timezone friendly dates.

### CODE REVIEW

File: lib/ai_helper/assistant.rb [hunk 1]
```diff
@@ -206,6 +229,8 @@ def icon_map(name)
           "question"
         when "illustrate_post"
           "images"
+        when "replace_dates"
+          "address-book"
```
[the remark this answers is not in the document]
Comment (SamSaffron) [reply lib/ai_helper/assistant.rb:233]: sure, sounds good!

File: db/fixtures/ai_helper/603_completion_prompts.rb [hunk 2]
```diff
@@ -217,3 +215,55 @@
...
+CompletionPrompt.seed do |cp|
+  cp.id = -310
+  cp.name = "replace_dates"
+  cp.prompt_type = CompletionPrompt.prompt_types[:diff]
+  cp.temperature = 0
+  cp.stop_sequences = ["\n</output>"]
+  cp.messages = {
+    insts: <<~TEXT,
+      You are a date and time formatter for Discourse posts. Convert natural language time references into date placeholders.
+      Do not modify any markdown, code blocks, or existing date formats.
```
[the remark this answers is not in the document]
Comment (SamSaffron) [reply db/fixtures/ai_helper/603_completion_prompts.rb:228]: this is going to be pretty impossible, the llm is in charge of this.

File: config/locales/server.en.yml [hunk 3]
```diff
@@ -233,6 +233,7 @@ en:
         custom_prompt: "Custom Prompt"
         explain: "Explain"
         illustrate_post: "Illustrate Post"
+        replace_dates: "Smart Dates"
```
[the remark this answers is not in the document]
Comment (SamSaffron) [reply config/locales/server.en.yml:236]: cool will change

**Comment by SamSaffron** [comment]
system spec feels expensive for this especially since helper is already covered... added a bit more internal coverage though

File: db/fixtures/ai_helper/603_completion_prompts.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (jjaffeux) [reply db/fixtures/ai_helper/603_completion_prompts.rb:228]: But how do you test the LLM call locally? We have no way to do this atm right?

**Pull request merged**